geps/gep-1619.md

```diff
@@ -62,14 +62,111 @@ Header-based stateful sessions are achieved by a backend or gateway providing an
+- If the gateway initiates the session, the backend will be presented with session attributes regardless if it enabled them.
 - If the backend initiates the session, the gateway should allow this and not force persistence connections, unless specifically configured to. The gateway may decode and alter the cookie established by the backend to achieve session persistence.
-- If the gateway initiates the session, the backend will be presented with session attributes regardless if it enabled them. 
 
-The client application also plays in a role in session initiation. If a client doesn't want a request to belong an existing session, it can remove the session cookie or identifier to signal the gateway or backend to recreate the session.
+In the next sections, we will take a closer look at the initiation of sessions in both the gateway and the backend.
+Please note that in the next sections, we are examining the various scenarios in which a session can be initiated. We
+are not prescribing specific implementations for session persistence. The intention is to understand the possibilities
+and behaviors related to session initiation while the [API](#api) section will provide more details on specific
+implementation details.
+
+#### Gateway Initiated Sessions
+
+This represents the most straightforward scenario for utilizing cookies. When a request is made, the gateway includes
+the `set-cookie` header in the final response, prompting the client to generate a cookie. This cookie is subsequently
+used in future requests, allowing the gateway to consistently choose the same upstream, establishing a persistent
+session.
+
+Here an example implementation of this process:
+```mermaid
+sequenceDiagram
+    actor C as Client
+    participant G as Gateway
+    participant B as Backend
+    C->>+G: Request Web Page
+    activate G
+    G->>+B: Request
+    B-->>-G: Response
+    G->>G: Add set-cookie header
+    G-->>-C: Response<br>[set-cookie]
+    Note right of G: [set-cookie] indicates a response<br> with a set-cookie header
+    C->>C: Create Cookie<br>from set-cookie header
+    Note right of C: [cookie] indicates a request<br> with a cookie
+    C->>+G: Request Web Page<br>[cookie]
+    G->>G: Consistent lookup of<br>server using cookie value
+    G->>+B: Request<br>[cookie]
+    B-->>-G: Response
+    G-->>-C: Response
+```
+
+#### Backend Initiated Sessions
+
+When dealing with backend-initiated sessions, the process becomes somewhat more complex. The gateway must modify the
+cookie's value to establish persistent sessions. However, in the most straightforward scenario, if the backend is not
+configured for session persistence, it should refrain from modifying the cookie. In this situation, the gateway should
+remain passive and simply forward the `set-cookie` header as it is.
+
+Here's an example of where the backend initiating a session, but the gateway does not modify the cookie:
```

Comment:
If the gateway does not modify the cookie it (the gateway) won't be able to send the future requests to the same backend and we are saying it's okay in this scenario? (possibly because the session info is encoded in the cookie so a different backend can very well process the request)

Reply:
My intention in this section was not prescription of how implementions should handle these scenarios, but more of an introduction of the possible scenarios of gateways and session persistence. Near the top of this paragraph I wrote [this](https://github.com/kubernetes-sigs/gateway-api/pull/2159/files/a377c3d7cf31ffc6c1daf1acea8275e9c39ac64f#diff-afa0e827962fe48152635217c6bc954020d5fec3578af4a9e5f744a2b4568fffR77-R78) because I know there can be confusion on trying to introduce an idea, and whether we are prescribing something or not:
```
Please note that in the next sections, we are examining the various scenarios in which a session can be initiated. We
are not prescribing specific implementations for session persistence. The intention is to understand the possibilities
and behaviors related to session initiation while the [API](#api) section will provide more details on specific
implementation details.
```
 So, to answer your question, I don't think we are saying anything yet. See [Cookie Rewriting](https://github.com/kubernetes-sigs/gateway-api/pull/2159/files/a377c3d7cf31ffc6c1daf1acea8275e9c39ac64f#diff-afa0e827962fe48152635217c6bc954020d5fec3578af4a9e5f744a2b4568fffR516) for the prescription part.

Reply:
I wouldn't prescribe HOW - but WHAT.

Cookie rewritting, adding new cookies, other techniques are implementation detail. What we need to prescribe (maybe as optional) is that gateway must have a mode where it doesn't force affinity but is watching for a cookie generated by backend, and when detected should enable affinity.